### PR TITLE
Fix a bad interaction between `(env ...)` and `(merlin)` in the workspace file

### PR DESCRIPTION
**Pull request by ghost**
The toplevel `(env ...)` was ignored when at least one context had `(merlin)`. This PR fixes this and adds a regression test.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ next
 - Fix #1107. `-opaque` wasn't correctly being added to modules without
   an interface. (#1108, fix #1107, @rgrinberg)
 
+- Fix a bug causing the toplevel `env` stanza in the workspace file to
+  be ignored when at least one contexts had `(merlin)` (#1114, @diml)
```
Comment (emillon) [inline CHANGES.md]: typo: "one context"

File: src/workspace.ml [hunk 2]
```diff
@@ -186,7 +186,7 @@ let t ?x ?profile:cmdline_profile () =
         Loc.fail (Context.loc ctx)
           "you can only have one context for merlin"
       | Opam { merlin = true; _ }, None ->
-        { merlin_context = Some name; contexts = ctx :: t.contexts; env = None }
+        { t with contexts = ctx :: t.contexts; merlin_context = Some name }
```
Comment (emillon) [inline src/workspace.ml]: What do you think about making this more explicit by extracting `set_merlin_context_name` and `add_context` functions?

Comment (diml) [reply src/workspace.ml]: Looking at this code again, it's too complicated for what it is really doing. I simplified it.

**Comment by diml** [comment]
I disabled the test as it seems hard to make it work in travis. The code is simpler now so this bug is less likely to re-occur.

**Review by emillon — APPROVED** [review]
Thanks!

**Pull request merged**